fluster/utils.py

```diff
@@ -161,6 +161,32 @@ def normalize_path(path: str) -> str:
     return path
 
 
+def compare_yuv_files(reference_yuv: str, test_yuv: str, tolerance: int = 2, keep_files: bool = False) -> bool:
+    """Compare two YUV files pixel by pixel with the assertion abs(pixel_a - pixel_b) <= tolerance"""
+    with open(reference_yuv, "rb") as ref_file, open(test_yuv, "rb") as test_file:
+        ref_data = ref_file.read()
```

Comment:
You shouldn't read the whole file in to memory, read the data un chunks of for example 1024 bytes